rule/rule.go

```diff
@@ -210,6 +209,17 @@ func (r *Rule) CompileURL() (*regexp.Regexp, error) {
 	return m.compiledURL, nil
 }
 
+// Replace searches the input string and replaces each match (with the rule's pattern)
+// found with the replacement text.
+func (r *Rule) Replace(input, replacement string) (string, error) {
+	re, err := r.compileRegexp()
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	return re.Replace(input, replacement, -1, -1)
```

Comment:
Just a nitpick: while this is equivalent to `ReplaceAllString` I think that `ReplaceAllString` is more readable :)